google-cloud-spanner-hibernate-dialect/src/main/java/com/google/cloud/spanner/hibernate/SpannerSqlAstTranslator.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2019-2023 Google LLC
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+
+package com.google.cloud.spanner.hibernate;
+
+import org.hibernate.engine.spi.SessionFactoryImplementor;
+import org.hibernate.sql.ast.tree.Statement;
+import org.hibernate.sql.ast.tree.predicate.LikePredicate;
+import org.hibernate.sql.exec.spi.JdbcOperation;
+
+/** We need a translator for the LIKE operator, as Cloud Spanner does not support ESCAPE clauses. */
+public class SpannerSqlAstTranslator<T extends JdbcOperation>
+    extends org.hibernate.dialect.SpannerSqlAstTranslator<T> {
+
+  public SpannerSqlAstTranslator(SessionFactoryImplementor sessionFactory, Statement statement) {
+    super(sessionFactory, statement);
+  }
+
+  @Override
```

Comment:
I couldn't find the base method that we are overriding here - https://github.com/hibernate/hibernate-orm/blob/3eb324f714977c8756e21b450fbc33e271fdea6e/hibernate-core/src/main/java/org/hibernate/dialect/SpannerSqlAstTranslator.java . Am I not looking at the right base class?

Reply:
The method is defined one layer higher up in the class hierarchy here: https://github.com/hibernate/hibernate-orm/blob/0ecd5d8a45f28131edd8dc5a18cfbe97d91f92fc/hibernate-core/src/main/java/org/hibernate/sql/ast/spi/AbstractSqlAstTranslator.java#L7272